src/components/forms/LoginForm.tsx

```diff
@@ -1,74 +1,87 @@
-import React, { useCallback, useContext, useState } from 'react'
+import React, { useCallback, useContext } from 'react'
 import { useNavigate } from 'react-router-dom'
 import { Button, Form } from 'react-bootstrap'
 import { TokenContext } from '../../context/token'
 import useTranslationHook from '../../hook/TranslationHook'
 import { login } from '../../services/api'
 import { setTokenLocalStorage } from '../../utils/helpers'
 import { FlashMessageContext, FlashMessageType } from '../../context/flashmessage'
+import { useForm } from 'react-hook-form'
+import { LoginFormType } from '../../types'
 
 const LoginForm = (): JSX.Element => {
-  const [username, setUsername] = useState<string>('')
-  const [password, setPassword] = useState<string>('')
   const { setToken } = useContext(TokenContext)
   const { addErrorMessage, addSuccessMessage } = useContext(FlashMessageContext)
   const navigate = useNavigate()
   const { t } = useTranslationHook()
-  const [error, setError] = useState<string>('')
-
-  const onUsernameChange = (e: React.ChangeEvent<HTMLInputElement>): void =>
-    setUsername(e.target.value)
-
-  const onPasswordChange = (e: React.ChangeEvent<HTMLInputElement>): void =>
-    setPassword(e.target.value)
+  const { register, handleSubmit, formState: { errors } } = useForm<LoginFormType>()
+  const minLengthPassword = 6 // Minimum length of the password required
+  const minLengthUsername = 2 // Minimum length of the username required
```

Comment:
Is this aligned with what the backend is expecting?

Reply:
the backend does not define min but only max 150 characters

Reply:
OK, is it not a bit weird to put some limitations frontend side while it can still be bypassed easily by talking directly to the backend?

Reply:
we can manage this on the backend I'll put in a ticket so I don't forget.

Reply:
ticket cote backend : https://github.com/mynotif/mynotif-backend/issues/145

Reply:
we close the ticket because the login form does not need to be validated reactively but only on submission